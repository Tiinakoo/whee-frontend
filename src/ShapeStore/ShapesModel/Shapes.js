import React from "react";
import { observer } from "mobx-react-lite";
import Flex from "../../components/shared/Flex/Flex";
import Icon from "../../components/shared/Icon/Icon";
import Heading from "../../components/shared/Heading/Heading";
import BodyText from "../../components/shared/BodyText/BodyText";
import Element from "../../components/shared/Element/Element";
import PrimaryButton from "../../components/shared/PrimaryButton/PrimaryButton";
import translations from "../../translations";
import getLanguage from "../../doings/getLanguage/getLanguage";

const localTranslation = translations[getLanguage()];

const Shapes = ({ shapesModel: { shapes } }) =>
  shapes.map(({ id, name, description, price, currency, addToCart }) => (
    <div key={id}>
      <Flex spaceBetween lineUpHorizontallyTo="md">
        <Flex lineUpHorizontally centeredVertically>
          <Element padding="md" box="md">
            <Icon iconName={name} />
          </Element>

          <Flex>
            <Heading>{name}</Heading>

            <BodyText>{description}</BodyText>
          </Flex>
        </Flex>

        <Flex>
          <Heading alignRight>
            {price} {currency}
          </Heading>

          <PrimaryButton onClick={addToCart} data-add-to-cart-test={id}>
            {localTranslation.addToCart}
          </PrimaryButton>
        </Flex>
      </Flex>
    </div>
  ));

export default observer(Shapes);
