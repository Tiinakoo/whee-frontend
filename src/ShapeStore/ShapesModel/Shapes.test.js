import Shapes from "./Shapes";
import React from "react";
import { mount } from "enzyme";

describe("Shapes", () => {
  let component;
  let shapesModelStub;

  beforeEach(() => {
    shapesModelStub = {
      shapes: [
        {
          id: "some-shape-id",
          name: "some-shape-name",
          description: "some-shape-description",
          price: 100,
          currency: "â‚¬",
          addToCart: expect.any(Function),
        },
      ],
    };

    component = mount(<Shapes shapesModel={shapesModelStub} />);
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });
});
