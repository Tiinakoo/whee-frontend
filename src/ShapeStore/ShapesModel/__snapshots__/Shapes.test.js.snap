// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapes renders 1`] = `
<Shapes
  shapesModel={
    Object {
      "shapes": Array [
        Object {
          "addToCart": [Function],
          "currency": "€",
          "description": "some-shape-description",
          "id": "some-shape-id",
          "name": "some-shape-name",
          "price": 100,
        },
      ],
    }
  }
>
  <div
    key="some-shape-id"
  >
    <Flex
      lineUpHorizontallyTo="md"
      spaceBetween={true}
    >
      <div
        className="flex flex__spaceBetween flex__lineUpHorizontallyToMdScreen"
      >
        <Flex
          centeredVertically={true}
          lineUpHorizontally={true}
        >
          <div
            className="flex flex__centeredVertically flex__lineUpHorizontally"
          >
            <Element
              box="md"
              padding="md"
            >
              <div
                className="element element__minWidth element__paddingMd element__boxMd"
              >
                <Icon
                  iconName="some-shape-name"
                />
              </div>
            </Element>
            <Flex>
              <div
                className="flex"
              >
                <Heading>
                  <p
                    className="heading heading__sizeMd"
                  >
                    some-shape-name
                  </p>
                </Heading>
                <BodyText>
                  <p
                    className="bodyText bodyText__fontBody bodyText__sizeMd"
                  >
                    some-shape-description
                  </p>
                </BodyText>
              </div>
            </Flex>
          </div>
        </Flex>
        <Flex>
          <div
            className="flex"
          >
            <Heading
              alignRight={true}
            >
              <p
                className="heading heading__sizeMd heading__alignRight"
              >
                100
                 
                €
              </p>
            </Heading>
            <PrimaryButton
              data-add-to-cart-test="some-shape-id"
              onClick={[Function]}
            >
              <Button
                attention={true}
                data-add-to-cart-test="some-shape-id"
                onClick={[Function]}
                variant="primary"
              >
                <button
                  className="button button__primary button__attention button__size"
                  data-add-to-cart-test="some-shape-id"
                  onClick={[Function]}
                  type="button"
                >
                  <BodyText
                    font="body"
                    size="sm"
                  >
                    <p
                      className="bodyText bodyText__fontBody bodyText__sizeSm"
                    >
                      Add to cart
                    </p>
                  </BodyText>
                </button>
              </Button>
            </PrimaryButton>
          </div>
        </Flex>
      </div>
    </Flex>
  </div>
</Shapes>
`;
